docs/orleans/whats-new-in-orleans.md

```diff
@@ -0,0 +1,550 @@
+
+### Serializing `record` types
+
+Members defined in a record's primary constructor have implicit ids by default, in other words Orleans supports serializing `record` types. This means that you cannot change the parameter order for an already deployed type, since that will break compatibility with previous versions of your application (in the case of a rolling upgrade) and with serialized instances of that type in storage and streams. Members defined in the body of a record type do not share identities with the primary constructor parameters. This means that you can and should start your members
+
+### Serialization best practices
+
+- ✅ **Do** give your types aliases using the `[Alias("my-type")]` attribute. Types with aliases can be renamed without breaking compatibility.
+- ❌ **Do not** change a `record` to a regular `class` or vice-versa. Records and classes are not represented in an identical fashion since records have primary constructor members in addition to regular members and therefore the two are not interchangeable.
+- ❌ **Do not** add new types to an existing type hierarchy for a serializable type. You must not add a new base class to an existing type. You can safely add a new subclass to an existing type.
+- ✅ **Do** start all member ids at zero for each type. Ids in a subclass and its base class can safely overlap. Note that both properties in the following example have ids equal to `0`.
+
+```csharp
+[GenerateSerializer]
+public class MyBaseClass
+{
+    [Id(0)]
+    public int MyBaseInt { get; set; }
+}
+
+[GenerateSerializer]
+public class MySubClass : MyBaseClass
+{
+    [Id(0)]
+    public int MyBaseInt { get; set; }
+}
+```
+
+- ✅ **Do** widen numeric member types as needed. You can widen `sbyte` to `short` to `int` to `long`.
+  * You can narrow numeric member types but it will result in a runtime exception if observed values cannot be represented correctly by the narrowed type. For example, `int.MaxValue` cannot be represented by a `short` field, so narrowing an `int` field to `short` can result in a runtime exception if such a value were encountered.
+- ❌ **Do not** change the signedness of a numeric type member. You must not change a member's type from `uint` to `int` or an `int` to a `uint`, for example.
+
+#### Surrogates for serializing foreign types
+
+Often times, you will want to pass types between grains which you do not have full control over. In these cases, it may be impractical to convert to and from some custom-defined type in your application code manually. Orleans offers a solution for these situations in the form of surrogate types. Surrogates are serialized in place of their target type and have functionality to convert to and from the target type. Here is an example of a foreign type and a corresponding surrogate and converter:
```

Comment:
```suggestion
Sometimes you may need to pass types between grains which you do not have full control over. In these cases, it may be impractical to convert to and from some custom-defined type in your application code manually. Orleans offers a solution for these situations in the form of surrogate types. Surrogates are serialized in place of their target type and have functionality to convert to and from the target type. Here is an example of a foreign type and a corresponding surrogate and converter:
```